### ISSUE TITLE
Ability to use one-time code for passwordless using hosted login pages

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#2589]
### Problem

For completeness of the passwordless feature, it'd be great to be able to use a code for passwordless rather than a magic link.

This also gets around all the issues with #629 .

### Solution

* user wants to log in using passwordless
* after submission, user is shown a page with N text boxes on it
* user gets a N digit code emailed
* user enters N digit code
* user is logged in

Some of this is already built (you can set the passwordless/magic link code to be N characters right now, and modify the passwordless template to show the characters instead of having a link). But the entry page would need to be built.

### Alternatives/workarounds

Use magic links.

### Additional context

Came up on a prospective customer's call.

Internal: https://inversoft.slack.com/archives/C051S8N8E/p1702679227972979

Example of this from a different provider.

<img width="789" alt="Screenshot 2023-12-15 at 4 06 51 PM" src="https://github.com/FusionAuth/fusionauth-issues/assets/91825/18460cf4-8095-4e81-a8c0-bbc691406ccf">

Best practices doc around this functionality worth reviewing: https://bradfrost.com/blog/post/the-ux-of-login-codes/

### Related
- https://github.com/FusionAuth/fusionauth-issues/issues/629
- https://github.com/FusionAuth/fusionauth-issues/issues/1552

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
This is possible today. The Passwordless API is de-coupled from email. But if you want to use FusionAuth as the transport, or use this feature in the FusionAuth hosted pages, you are limited to email.

**Comment by robotdan** [comment]
I think this is mostly a duplicate of https://github.com/FusionAuth/fusionauth-issues/issues/1552?

**Comment by mooreds** [comment]
I don't think so, @robotdan . #1552 seems to be about the delivery method (SMS) whereas this issue is about the code entry method (clicking a link [current] vs entering a 4-6 digit number [what this issue discusses]).

**Comment by robotdan** [comment]
I think this will be possible by way of changes made for https://github.com/FusionAuth/fusionauth-issues/issues/1.

**Comment by mrudatsprint** [comment]
This is now possible for a phone number identity with the changes made for https://github.com/FusionAuth/fusionauth-issues/issues/1.

However, email identity only supports a clickable link vs entering a 4-6 digit number that was sent via email.

An idea would be for passwordless to have a Verification Strategy much like the `Tenant->Identities->Identity verification settings->Verification Strategy` of `Clickable link` or `Form field`
